### PR TITLE
feat: added modal component

### PR DESCRIPTION
**Pull request by pradipgharge**

### CODE REVIEW

**Review by AshutoshM29 — COMMENTED** [review]
Nice work buddy!!
UI looks nice, can be merged

**Review by sainath7878 — APPROVED** [review]
LGTM!

File: pages/components/Modal/modal.html [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    </p>
+                </div>
+                <div class="component-variants">
+                    <h3 class="component-varaint-heading">Modal Example</h3>
+                </div>
+                <div class="modal-preview-container">
+                    <div class="modal">
+                        <div class="modal-header">
+                            <h2 class="modal-heading">Modal Title</h2>
+                            <span class="btn-dismiss">X</span>
```
Comment (manish-GH) [inline pages/components/Modal/modal.html]: It's probably better to use x-icon here. Also give it a color other than white, I needs to be visible even if the user is not hovering over it

**Review by prankurpandeyy — APPROVED** [review]

**Comment by prankurpandeyy** [comment]
awesome work , you can merge

**Review by Sahil25061999 — APPROVED** [review]
In the modal.css since the display: flex and some other proerties are being repeated maybe you can give it in the same line using " , " separator

**Review by manish-GH — CHANGES_REQUESTED** [review]
Add modal container to your code snippet. Also if you have the time you could consider adding a button that displays the modal on click

File: pages/components/Modal/modal.html (hunk 1 above)
Comment (pradipgharge) [reply pages/components/Modal/modal.html]: Made the changes !

**Comment by pradipgharge** [comment]
> In the modal.css since the display: flex and some other proerties are being repeated maybe you can give it in the same line using " , " separator

Hey Sahil, keeping it that way for code clarity.

File: pages/components/Modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    padding: 1rem;
+    justify-content: space-evenly;
+    align-items: center;
+    gap: 1rem;
+}
+
+.modal{
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
```
Comment (amartya-rs) [inline pages/components/Modal/modal.css]: The default value of flex-wrap is nowrap. I guess you can omit this line.

Comment (pradipgharge) [reply pages/components/Modal/modal.css]: Ok thanks made the changes

**Review by amartya-rs — APPROVED** [review]

**Review by manish-GH — APPROVED** [review]

**Pull request merged**